### PR TITLE
Homework3: Natalia Mashliakevich

### PR DESCRIPTION
**Pull request by nataliasipenko**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle).
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks.
- [ ] my PR made from the fork feature branch to appropriate branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/homework3/task1.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""
+FizzBuzz
+
+Напишите программу, которая печатает цифры от 1 до 100, но вместо чисел,
+кратных 3 пишет Fizz, вместо чисел кратный 5 пишет Buzz, а вместо чисел
+одновременно кратных и 3 и 5 - FizzBuzz
+"""
+
+for number in range(1, 101):
+    if number % 3 == 0 and number % 5 == 0:
```
Comment (belovmd) [inline src/homework3/task1.py]: 15?

Comment (belovmd) [reply src/homework3/task1.py]: `not smth` instead of `smth == 0`

File: src/homework3/task2.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+4 Одной строкой удалите элемент  '2a' из прошлого списка и напечатайте его.
+5 Скопируйте список и добавьте в него элемент '2a' так чтобы в исходном списке этого элемента
+не было.
+"""
+
+import copy
+
+task1 = [a + b for a in "ab" for b in "bcd"]
+task2 = task1[::2]
+task3 = [c + d for c in "1234" for d in "a"]
```
Comment (belovmd) [inline src/homework3/task2.py]: + 'a'

File: src/homework3/task4.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+Входные данные - строка из чисел, разделенная пробелами.
+Выходные данные - количество пар.
+Важно: 1 1 1 - это 3 пары, 1 1 1 1 - это 6 пар
+"""
+
+str1 = '1 1 1 1'
+str2 = ''.join(str1.split())
+pair_counter = 0
+for i in range(len(str2)):
+    for j in range(i + 1, len(str2)):
```
Comment (belovmd) [inline src/homework3/task4.py]: can be solved with O(n)

File: src/homework3/task6.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+"""
+Упорядоченный список.
+
+Дан список целых чисел. Требуется переместить все ненулевые элементы в левую часть списка,
+не меняя их порядок, а все нули - в правую часть. Порядок ненулевых элементов изменять нельзя,
+дополнительный список использовать нельзя, задачу нужно выполнить за один проход по списку.
+Распечатайте полученный список.
+"""
+
+# Через while-loop
```
Comment (belovmd) [inline src/homework3/task6.py]: please choose one.

**Review by belovmd — CHANGES_REQUESTED** [review]
- sanity checks failed - redundant code in PR

**Review by nataliasipenko — COMMENTED** [review]
Changes made. Review requested. 
Thank you in advance!

File: src/homework3/task2.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+5 Скопируйте список и добавьте в него элемент '2a' так чтобы в исходном списке этого элемента
+не было.
+"""
+
+import copy
+
+task1 = [a + b for a in 'ab' for b in 'bcd']
+task2 = task1[::2]
+task3 = [c + d for c in '1234' for d in 'a']
+print(task3.pop(1))  # task4
```
Comment (belovmd) [inline src/homework3/task2.py:17]: redundant comment

File: src/homework3/task2.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+4 Одной строкой удалите элемент  '2a' из прошлого списка и напечатайте его.
+5 Скопируйте список и добавьте в него элемент '2a' так чтобы в исходном списке этого элемента
+не было.
+"""
+
+import copy
+
+task1 = [a + b for a in 'ab' for b in 'bcd']
+task2 = task1[::2]
+task3 = [c + d for c in '1234' for d in 'a']
```
Comment (belovmd) [inline src/homework3/task2.py:16]: c + 'a'

**Review by belovmd — APPROVED** [review]
8

**Pull request closed without merging**